pkgdb/src/buildenv/realise.cc

```diff
@@ -561,11 +561,9 @@ static std::vector<std::pair<std::string, resolver::LockedPackageRaw>>
 getLockedPackages( const resolver::LockfileRaw & lockfile,
                    const System &                system )
 {
-  traceLog( "getting locked packages" );
-  auto packages = lockfile.packages.find( system );
-  if ( packages == lockfile.packages.end() )
+  auto systems = lockfile.manifest.getSystems();
+  if ( std::find( systems.begin(), systems.end(), system ) == systems.end() )
     {
-      // Custom exception for non supported system
       throw SystemNotSupportedByLockfile(
         "'" + system + "' not supported by this environment" );
     }
```

Comment:
I think this is semantically different. I.e.

```toml
[install]
mypkg = {path = "mypkg", systems = "aarch64-darwin"}

[options]
systems = [ "x86_64-linux" ]
```

used to pass before but doesnt seem to pass anymore if i see that right

Reply:
On main:
```
> cat /tmp/manifest
[install]
hello = {pkg-path = "hello", systems = ["x86_64-linux"]}

[options]
systems = [ "aarch64-darwin" ]
> flox edit -f /tmp/manifest
❌ ERROR: descriptor `install.hello' specifies system `x86_64-linux' which is not in `options.systems' in the manifest.
```

Reply:
huh, i guess that's in another path ^^ (check maybe?)
didn't know that's a constraint